### ISSUE TITLE
Setup uiowa04

### ISSUE THREAD

**Issue by joewhitsitt** [uiowa/uiowa#5522]
Didn't see another issue. Keep forgetting it exists.

Possible intersection with PHP upgrade https://github.com/uiowa/uiowa/issues/5315#issuecomment-1147565179

# Proposed Solution

- [x] Follow docs here https://wiki.uiowa.edu/display/drupaldev/Acquia+Applications
- [x] Add uihealthcarecorp to metadata generator

**Comment by joewhitsitt** [comment]
Rick created a wiki entry for this before he left: https://wiki.uiowa.edu/display/drupaldev/Acquia+Applications

Easy peasy?

**Comment by joewhitsitt** [comment]
We will also need to review the cron offsets

**Comment by makurj** [comment]
I'd like to drive/pair on this.

**Comment by makurj** [comment]
The old wiki doc was moved to https://wiki.uiowa.edu/display/drupaldev/Acquia+Cloud

**Comment by bspeare** [comment]
I'm planning on pairing with @makurj on this today around 2:30.

**Comment by briand44** [comment]
One thing to keep in mind here is if we need to set this up for the metadata generator, there is a card in PL to do this for uihealthcarecorp as well. Maybe makes sense to add that while working on this? https://github.com/orgs/uiowa/projects/1#card-86437297

**Comment by bspeare** [comment]
Yes, I think it makes sense to fold that work in.

**Comment by joewhitsitt** [comment]
I wonder if this should coincide with a new release/clean main since the doc talks about "the branches in the base application must exist in the same state"

**Comment by bspeare** [comment]
@joewhitsitt that does sound important.  Can you spell out what we need to do to get to that same state?

**Comment by joewhitsitt** [comment]
I was just thinking if @pyrello is doing a provisions release today that would get things in sync across all apps.

**Comment by pyrello** [comment]
Working on it

**Comment by bspeare** [comment]
So, once the release is finished we can proceed with setting this up?  Maybe tomorrow morning might be a good starting point for this work.

**Comment by makurj** [comment]
Waiting for the next successful deployment to start work on this.

**Comment by joewhitsitt** [comment]
deployment went yesterday (the config diff issue doesn't mess with this currently): https://github.com/uiowa/uiowa/releases/tag/3.19.7

**Comment by joewhitsitt** [comment]
holding again for https://github.com/uiowa/uiowa/releases/tag/3.19.8 to go out

**Comment by joewhitsitt** [comment]
I think i figured it out. Moving forward a bit and updating documentation

**Comment by joewhitsitt** [comment]
>Next, set up the new application in the BLT config, Travis config and the custom UIowa BLT config

I held off on this for now. We can pick up with that tomorrow.

**Comment by joewhitsitt** [comment]
We aren't done yet.

**Comment by joewhitsitt** [comment]
```
To svn-37631.prod.hosting.acquia.com:uiowa04.git
 ! [rejected]          main-build -> main-build (fetch first)
error: failed to push some refs to 'svn-37631.prod.hosting.acquia.com:uiowa04.git'
hint: Updates were rejected because the remote contains work that you do
hint: not have locally.
```
Looking to resolve this.

**Comment by joewhitsitt** [comment]
This is not going well. uiowa04 has that md5 hash now, but travis is complaining. in an effort to resolve, I think I just compounded the problem

**Comment by makurj** [comment]
I just saw it fail a second time :( What's our next step?

**Comment by joewhitsitt** [comment]
🤷

**Comment by joewhitsitt** [comment]
https://app.travis-ci.com/github/uiowa/uiowa/builds/257308661

My guess is that I didn't properly reset yesterday

Update: Also got develop-build working https://app.travis-ci.com/github/uiowa/uiowa/builds/257316657

**Comment by makurj** [comment]
I don't have access so if someone with access could:

1. Change the scheduled job for DEV, Stage and Prod to the correct time.
2. Check Search and Variables for all enviroments.
3. Email settings under Config > Advanced.
4. Any CDN settings.
5. Log forwarding.
6. Team perms.

Left out brand and v2 scheduled jobs.

**Comment by joewhitsitt** [comment]
>There are no associated domains for this environment's application.

Think we need the default site or _a_ site to get platform email enabled

**Comment by makurj** [comment]
@briand44 We need new Relic keys (API key and App ID) for uiowa04.

**Comment by briand44** [comment]
>We need new Relic keys (API key and App ID) for uiowa04.

Can't you get the API key from a different application. App ID would be uiowa04.prod

**Comment by makurj** [comment]
That's not what we found. Can you join the #drupal huddle?

**Comment by makurj** [comment]
Next step: Log into New Relic and create new info for uiowa04 to add to uiowa04 variables.

**Comment by joewhitsitt** [comment]
I think the env variables for new relic relate to https://docs.newrelic.com/docs/apm/new-relic-apm/maintenance/record-monitor-deployments/. `hooks/prod/post-code-deploy/newrelic.sh` I don't have access to this area of New Relic so I think we will need @briand44 (admin) to investigate further.

**Comment by makurj** [comment]
New Relic, authentication and default site install completed.

**Comment by makurj** [comment]
@mark-bennett-uiowa You can go ahead with the sites you wanted on uiowa04. Once that is done, @joewhitsitt can enable Platform email on uiowa04.

**Comment by makurj** [comment]
Huge thanks to @joewhitsitt and @briand44!
